### PR TITLE
@sarahscott => Use gemini for large image thumbnails when possible

### PR DESCRIPTION
**Pull request by orta**
Creates a `GeminiThumbnailable` protocol which cares about whether something has a gemini token, then adds some extra functions to create thumbnails based on a max width/height.

### CODE REVIEW

File: Emergence/Models/Artsy Model Protocols/GeminiThumbnailable.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+let geminiServerAddress = "d7hftxdivxxvm.cloudfront.net"
+
+protocol GeminiThumbnailable {
+    var geminiToken: String? { get }
+}
+
+extension GeminiThumbnailable {
+
+    /// Returns a NSURL represnting a thumbnail using Gemini to process the image
```
Comment (sarahscott) [inline Emergence/Models/Artsy Model Protocols/GeminiThumbnailable.swift:15]: representing*

File: Emergence/Models/Electric Objects/Image.swift [hunk 2]
```diff
@@ -2,14 +2,22 @@ import Gloss
...
     let aspectRatio: CGFloat?
+    let geminiToken: String?
 
     let isDefault: Bool
+
+    func bestThumbnailWithHeight(height:CGFloat) -> NSURL? {
+        if let url = geminiThumbnailURLWithHeight(Int(height)) {
+            return url
+        }
+        return bestAvailableThumbnailURL()
```
Comment (sarahscott) [inline Emergence/Models/Electric Objects/Image.swift:19]: Where is this declared?

Comment (orta) [inline Emergence/Models/Electric Objects/Image.swift:19]: its a part of the `Thumbnailable` protocol : https://github.com/artsy/Emergence/blob/master/Emergence/Models/Artsy%20Model%20Protocols/Thumbnailable.swift

**Comment by orta** [comment]
:+1:  Fixed the typo in the next PR

**Pull request merged**